### PR TITLE
fix(RDF API): Incorrect internal IRIs

### LINKED ISSUE #4382: RDF API output generates incorrect internal IRIs
**What version of EMX2 are you using (see footer)**
v11.22.0-SNAPSHOT (when tested on localhost, though dev server also shows issue)

**Describe the bug**
The RDF generates invalid internal IRIs.

**To Reproduce**
1. Run `curl https://emx2.dev.molgenis.org/dcatje/api/rdf/Distribution?name=Analyses`

**Expected behavior**
Internal IRIs should be identical to [v11.16.2](https://github.com/molgenis/molgenis-emx2/releases/tag/v11.16.2) and older. For example: 
```
http://emx2.dev.molgenis.org//dcatje/api/rdf/Distribution/dcatje/api/rdf/Distribution/column/name
```
should be
```
http://emx2.dev.molgenis.org/dcatje/api/rdf/Distribution/column/name
```

**Additional context**
Bug might have been introduced with #4165?

### PR DESCRIPTION
**Pull request by svandenhoek**
What are the main changes you did:
- Some changes due to #4165:
  - Removed unused return values
  - Refactored code to better make use of `Context`
- Closes #4382: `RDFService` expects base URL of installation only:
  - https://github.com/molgenis/molgenis-emx2/blob/9b22e280e06b79b0437d88a46a2afcd4f71f3713/backend/molgenis-emx2-rdf/src/main/java/org/molgenis/emx2/rdf/RDFService.java#L100-L102
  - https://github.com/molgenis/molgenis-emx2/blob/9b22e280e06b79b0437d88a46a2afcd4f71f3713/backend/molgenis-emx2-rdf/src/main/java/org/molgenis/emx2/rdf/RDFService.java#L109

how to test:
- See if behaviour of #4382 does not occur anymore
- Check if URL still correct regarding presence/absence port:
  - On local server: `curl http://localhost:8080/pet%20store/api/rdf` should contain `<http://localhost:8080/pet%20store/api/rdf/User/column/username>`
  - On preview server: `curl https://preview-emx2-pr-4383.dev.molgenis.org/pet%20store/api/rdf` should contain `<http://preview-emx2-pr-4383.dev.molgenis.org/pet%20store/api/rdf/User/column/username>`

todo:
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

### CODE REVIEW

**Review by harmbrugge — COMMENTED** [review]
I think using contextPath wil fix the issue, but indeed a test would be nice

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/RDFApi.java [hunk 1]
```diff
@@ -5,10 +5,7 @@
 import io.javalin.Javalin;
 import io.javalin.http.Context;
 import java.io.*;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (svandenhoek) [reply backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/RDFApi.java:8]: IDEA did that automatically after I made some changes, I'll make sure to revert it.

**Comment by svandenhoek** [comment]
Seems `config.router.contextPath` (as stated in the [Javalin docs](https://javalin.io/documentation#routerconfig)) is currently not being configured in:
https://github.com/molgenis/molgenis-emx2/blob/a5e83107f663b3b0f3fe38ed4eebd84ffa38a6de/backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisWebservice.java#L43-L52

So the current fix might not work yet anyhow (or any other places that depend on a defined `contextPath`)...

**Comment by svandenhoek** [comment]
Tests were added where `config.contextResolver.host` & `config.router.contextPath` are set to test the behaviour. In my initial feedback processing (510b1b57a88ed547317706a5229635fc99800137) Javalin actually used the said port (and locally all tests worked), but the Azure build failed due to `java.net.BindException: Permission denied` on port 80.

So currently using the Javalin test framework that uses a random uncommon port and simply overriding the defined `host`. This means some of the tests were removed though that used the actual port.

**Review by harmbrugge — APPROVED** [review]

**Review by esthervanenckevort — APPROVED** [review]

**Pull request merged**